src/foam/u2/view/OverlayActionListView.js

```diff
@@ -143,18 +147,16 @@ foam.CLASS({
         self.overlay_.close();
       });
 
+      // a list where element at i stores whether ith action in data is available or not
+      const showActions = await Promise.all(this.data.map(this.showAction.bind(this)));
       this.overlay_.startContext({ data: self.obj })
-        .forEach(self.data, function(action) {
-          if ( action.createIsAvailable$(self.__context__, self.obj).value ) {
+        .forEach(self.data, function(action, index) {
+          if ( showActions[index] ) {
             this
               .start()
                 .addClass(self.myClass('button-container'))
-                .addClass(action.createIsEnabled$(self.__context__, self.obj).map( e => ! e && self.myClass('disabled')))
```

Comment:
@sarthak-marwaha I mistook `createIsEnabled` as `createIsAvailable`. This shouldn't be removed, right?

Reply:
Yeah this should stay.